### PR TITLE
add where clause

### PR DESCRIPTION
**Pull request by AjayTripathy**
Add a simple clause to reduce amount of data returned to the server. Regression tested with AWS external cost tests.

### CODE REVIEW

File: pkg/cloud/awsprovider.go [hunk 1]
```diff
@@ -1290,8 +1290,8 @@ func (a *AWS) ExternalAllocations(start string, end string, aggregators []string
 			%s,
 			SUM(line_item_blended_cost) as blended_cost
 		FROM %s as cost_data
-		WHERE (%s='%s') AND line_item_usage_start_date BETWEEN date '%s' AND date '%s'
-		GROUP BY %s`, aggregatorNames, filter_column_name, customPricing.AthenaTable, filter_column_name, filterValue, start, end, groupby)
+		WHERE (%s='%s') AND line_item_usage_start_date BETWEEN date '%s' AND date '%s AND %s <> ''
```
Comment (nikovacevic) [inline pkg/cloud/awsprovider.go:1293]: Are the quotes correct here? Seems like it should be `... AND date '%s' AND...` (with a close quote on the %s) but perhaps I'm just reading this wrong.

Comment (nikovacevic) [reply pkg/cloud/awsprovider.go:1293]: And why only use `formattedAggregators[0]`?

Comment (AjayTripathy) [reply pkg/cloud/awsprovider.go]: Fixed, thanks. It turns out I was testing the wrong if clause...

Comment (nikovacevic) [reply pkg/cloud/awsprovider.go]: Ah that makes sense. This one looks better :+1:

**Pull request merged**